[ja] Translate "Completing the largest migration in Kubernetes history" into Japanese (#46802)
kubernetes/website: content/ja/blog/_posts/2024-05-20-cloud-provider-migration.md

---
layout: blog

> title: 'Kubernetes史上最大の移行を完了'

Reviewer: ```suggestion
title: 'Kubernetes史上最大の移行作業を完了'
```

移行だけだと意味がよくわからん

Author: 確かに、今のタイトルだと具体的に何しているのかわかりませんよね。修正します。

date: 2024-05-20
slug: completing-cloud-provider-migration
---

**著者:** Andrew Sy Kim (Google), Michelle Au (Google), Walter Fender (Google), Michael McCune (Red Hat)

> **翻訳者:** Taisuke Okamoto (IDC Frontier Inc),, [Junya Okabe](https://github.com/Okabe-Junya) (University of Tsukuba)

Reviewer 2: ```suggestion
**翻訳者:** Taisuke Okamoto (IDC Frontier Inc), [Junya Okabe](https://github.com/Okabe-Junya) (University of Tsukuba)
```

nit: `,` が2つ入っていそうです

Kubernetes v1.7以降、Kubernetesプロジェクトは、クラウドプロバイダーとの統合機能をKubernetesのコアコンポーネントから分離するという野心的な目標を追求してきました([KEP-2395](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers/README.md))。

> この統合機能はKubernetesの初期の開発と成長に重要な役割を果たしつつも、２つの重要な要因によってその分離が推進されました。

Author: @Okabe-Junya 
改めて、考え直してみました。

- 「削除」という表現は不自然かなと感じたので、「取り除く」という表現に変更しました
- 「クラウドプロバイダーとの統合機能を取り除く」だけでは、文章としてつながらないなと思い、「Kubernetesのコアコンポーネントから」という文言を追加しました

> While these integrations were instrumental in Kubernetes' early development and growth, their removal was driven by two key factors

- 直訳すると「これら」、「それら」になりますが、前段の文章で「クラウドプロバイダーとの統合機能をKubernetesのコアコンポーネントから取り除く」と日本語訳したので、単一の概念として表しているため、単数形で表現しました

Reviewer 2: ありがとうございます！SGTMです！

2. **APIサーバーネットワークプロキシ** ([KEP-1281](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1281-network-proxy))
3. **kubeletクレデンシャルプロバイダープラグイン** ([KEP-2133](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2133-kubelet-credential-providers))
4. **[CSI](https://github.com/container-storage-interface/spec?tab=readme-ov-file#container-storage-interface-csi-specification-)を使用するストレージの移行** ([KEP-625](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/625-csi-migration/README.md))

各サブシステムは、組み込み機能と同等の機能を実現するために不可欠であり、安全で信頼できる移行パスを使用して各サブシステムをGAレベルの成熟度にするために、いくつかのリリースが必要でした。
以下に、各サブシステムの詳細を説明します。

### クラウドコントローラーマネージャー

クラウドコントローラーマネージャーは、この取り組みで導入された最初の外部コンポーネントであり、`kube-controller-manager`と`kubelet`のうち、クラウドAPIと直接やり取りする機能を置き換えるものです。
この重要なコンポーネントは、ノードが実行されているクラウドのリージョンとゾーンを示すメタデータラベルや、クラウドプロバイダーのみが知っているIPアドレスを適用することにより、ノードを初期化する役割を担っています。
さらに、LoadBalancerタイプのServiceに対してクラウドロードバランサーをプロビジョニングするサービスコントローラーも実行します。

![Kubernetesのコンポーネント](/images/docs/components-of-kubernetes.svg)

> 詳細については、Kubernetesドキュメントの[クラウドコントローラーマネージャー](/docs/concepts/architecture/cloud-controller/)を参照してください。

Reviewer 2: ```suggestion
詳細については、Kubernetesドキュメントの[クラウドコントローラーマネージャー](/ja/docs/concepts/architecture/cloud-controller/)を参照してください。
```

APIサーバーネットワークプロキシの詳細については、[README](https://github.com/kubernetes-sigs/apiserver-network-proxy#readme)を参照してください。

### kubeletのクレデンシャルプロバイダープラグイン

`kubelet`のクレデンシャルプロバイダープラグインは、Google Cloud、AWS、またはAzureでホストされているイメージレジストリのクレデンシャルを動的に取得する`kubelet`の組み込み機能を置き換えるために開発されました。
従来の機能は便利で、`kubelet`がGCR、ECR、またはACRからイメージをプルするための短期間のトークンをシームレスに取得できるようにしていました。
しかし、Kubernetesの他の領域と同様に、これをサポートするには、`kubelet`が異なるクラウド環境とAPIについて特定の知識を持つ必要がありました。

2019年に導入されたクレデンシャルプロバイダープラグインメカニズムは、`kubelet`が様々なクラウドでホストされているイメージのクレデンシャルを動的に提供するプラグインバイナリを実行するための汎用的な拡張ポイントを提供します。
この拡張性により、`kubelet`の短期間のトークンを取得する機能が、最初の3つのクラウドプロバイダーを超えて拡張されました。

詳細については、[認証されたイメージプルのためのkubeletクレデンシャルプロバイダー](/docs/concepts/containers/images/#kubelet-credential-provider)を参照してください。

### ストレージプラグインのKubernetesコアからCSIへの移行

Container Storage Interface(CSI)は、Kubernetesやそのほかのコンテナオーケストレーターにおいてブロックおよびファイルストレージシステムを管理するためのコントロールプレーン標準であり、1.13でGAになりました。
これは、Kubernetesに直接組み込まれていたボリュームプラグインを、Kubernetesクラスター内のPodとして実行できるドライバーに置き換えるために設計されました。
これらのドライバーは、Kubernetes APIを介して`kube-controller-manager`ストレージコントローラーと通信し、ローカルのgRPCエンドポイントを介して`kubelet`と通信します。
現在、すべての主要なクラウドとストレージベンダーにわたって100以上のCSIドライバーが利用可能であり、Kubernetesでステートフルなワークロードが現実のものとなっています。

> ただし、Kubernetesコアのボリューム APIの既存のすべてのユーザーをどのように扱うかという大きな課題が残っていました。

Reviewer 2: ```suggestion
ただし、KubernetesコアのボリュームAPIの既存のすべてのユーザーをどのように扱うかという大きな課題が残っていました。
```

APIの後方互換性を維持するために、Kubernetesコアのボリューム APIを同等のCSI APIに変換するAPIトランスレーション層をコントローラーに組み込みました。
これにより、すべてのストレージ操作をCSIドライバーにリダイレクトすることができ、APIを削除せずにKubernetesコアのボリュームプラグインのコードを削除する道が開けました。

> Kubernetesコアのストレージの移行の詳細については、「[Kubernetesコアからのボリュームの移行がCSIでベータ版になりました](https://kubernetes.io/blog/2019/12/09/kubernetes-1-17-feature-csi-migration-beta/)」を参照してください。

Reviewer 2: ```suggestion
Kubernetesコアのストレージの移行の詳細については、[Kubernetes In-Tree to CSI Volume Migration Moves to Beta](https://kubernetes.io/blog/2019/12/09/kubernetes-1-17-feature-csi-migration-beta/)を参照してください。
```

IMHO: either is fine な気もしますが、未翻訳のブログなのでタイトルは英語ママで良い気がしました。合わせて`「」`は削除して良いように感じました

Reviewer 3: そうですね、確かに未翻訳なので英語表記のままにしておきます。
また、「」に関しては、通常のドキュメントを拝見していてもついていないので削除します。

## 今後の展望

> この移行は、ここ数年のSIG Cloud Providerの主要な焦点でした。

Reviewer 2: ```suggestion
この移行は、ここ数年のSIG Cloud Providerがもっとも注力してきたことでした。
```

IMHO

Reviewer 3: たしかに、「注力する」の方が言い回し的には良いと思います。そちらを採用させていただきます！

この重要なマイルストーンを達成したことで、これまでに構築してきた外部サブシステムを活用して、Kubernetesとクラウドプロバイダーをより良く統合するための新しい革新的な方法を模索する取り組みにシフトしていきます。
これには、クラスター内のノードがパブリッククラウドとプライベートクラウドの両方で実行できるハイブリッド環境でKubernetesをより賢くすることや、外部プロバイダーの開発者が統合の取り組みを簡素化・合理化するためのより良いツールとフレームワークを提供することが含まれます。

> 計画されているすべての新機能、ツール、フレームワークを含め、SIG Cloud Providerは方程式のもう一方の側面であるテストを忘れてはいません。

Reviewer 2: ```suggestion
新機能、ツール、フレームワークの計画が進む中、SIG Cloud Providerはテストの重要性も忘れていません。
```

IMO: 意訳っぽいですが、こんな感じでどうでしょうか

SIGの将来の活動のもう1つの重点分野は、より多くのプロバイダーを含めるためのクラウドコントローラーテストの改善です。
この取り組みの最終目標は、できるだけ多くのプロバイダーを含むテストフレームワークを作成し、Kubernetesコミュニティに対して、Kubernetes環境に関する最高レベルの信頼性を提供することです。

> v1.29より前のバージョンのKubernetesを使用していて、まだ外部クラウドプロバイダーに移行していない場合は、以前のブログ記事「[Kubernetes 1.29: クラウドプロバイダー統合が独立したコンポーネントになりました](/blog/2023/12/14/cloud-provider-integration-changes/)」を確認することをおすすめします。

Reviewer 2: ditto

> IMHO: either is fine な気もしますが、未翻訳のブログなのでタイトルは英語ママで良い気がしました。合わせて`「」`は削除して良いように感じました
